efel/pyfeatures/extrafeats.py

```diff
@@ -0,0 +1,521 @@
+        max_idx = np.min([max_idx, waveforms.shape[1]])
+
+        if len(time[peak_idx[i]:max_idx]) < 3:
+            continue
+        slope = _get_slope(
+            time[peak_idx[i]:max_idx], waveforms[i, peak_idx[i]:max_idx]
+        )
+        rslope[i] = slope[0]
+
+    return rslope
+
+
+def peak_image(waveforms, sign="negative"):
+    """
+    Normalized amplitude at the time of minimum or maximum peak.
+
+    Parameters
+    ----------
+    waveforms  : numpy.ndarray (num_waveforms x num_samples)
+        waveforms to compute features for
+    sign : str
+        "pos" | "neg"
+
+    Returns
+    -------
+    np.ndarray
+        Peak images for the waveforms
+
+    """
+    assert len(waveforms) > 1
+
+    if sign == "negative":
+        funarg = np.argmin
+        fun = np.min
+    else:
+        funarg = np.argmax
+        fun = np.max
+
+    peak_channel, peak_time = np.unravel_index(
+        funarg(waveforms), waveforms.shape
+    )
+    relative_peaks = waveforms[:, peak_time] / fun(waveforms[peak_channel])
+
+    return relative_peaks
+
+
+def relative_amplitude(waveforms, sign="negative"):
+    """
+    Normalized amplitude with respect to channel with largest amplitude.
+
+    Parameters
+    ----------
+    waveforms  : numpy.ndarray (num_waveforms x num_samples)
+        waveforms to compute features for
+    fs : float
+        Sampling rate in Hz
+    sign : str
+        "positive" | "negative"
+
+    Returns
+    -------
+    np.ndarray
+        Relative amplitudes for the waveforms
+
+    """
+    assert len(waveforms) > 1
+
+    if sign == "negative":
+        fun = np.min
+    else:
+        fun = np.max
+
+    peak_amp = np.abs(fun(waveforms))
+    relative_peaks = np.abs(fun(waveforms, 1)) / peak_amp
+
+    return relative_peaks
+
+
+def peak_time_diff(waveforms, fs, sign="negative"):
+    """
+    Peak time differences with respect to channel with largest amplitude.
+
+    Parameters
+    ----------
+    waveforms  : numpy.ndarray (num_waveforms x num_samples)
+        waveforms to compute features for
+    fs : float
+        Sampling rate in Hz
+    sign : str
+        "positive" | "negative"
+
+    Returns
+    -------
+    np.ndarray
+        Peak time differences for the waveforms
+
+    """
+    assert len(waveforms) > 1
+
+    if sign == "negative":
+        argfun = np.argmin
+    else:
+        argfun = np.argmax
+
+    peak_chan = np.unravel_index(argfun(waveforms), waveforms.shape)[0]
+    peak_time = argfun(waveforms[peak_chan])
+    relative_peak_times = (argfun(waveforms, 1) - peak_time) / fs
+
+    return relative_peak_times
+
+
+def _get_slope(x, y):
+    """
+    Retrun the slope of x and y data, using scipy.signal.linregress
+    """
+    from scipy.stats import linregress
+
+    slope = linregress(x, y)
+    return slope
+
+
+def _get_trough_and_peak_idx(waveform, after_max_trough=False):
+    """
+    Return the indices into the input waveforms of the detected troughs
```

Comment:
I think `the indices of the detected troughs (minimum of waveform) and peaks (maximum of waveform, after trough) of the input waveforms` is clearer